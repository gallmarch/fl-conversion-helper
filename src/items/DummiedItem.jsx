import React, { Component } from 'react';
import PropTypes from 'prop-types';

import ToolTip from '../tooltips/ToolTip';
import { IMAGE_ROOT } from './Item';

export default class DummiedItem extends Component {
  constructor(props) {
    super(props);
    this.handleMouseLeave = this.handleMouseLeave.bind(this);
    this.handleMouseMove = this.handleMouseMove.bind(this);
    this.state = { showToolTip: false };
  }

  handleMouseLeave() {
    this.setState({ showToolTip: false });
  }

  handleMouseMove() {
    this.setState({ showToolTip: true });
  }

  render() {
    const { data } = this.props;
    const { showToolTip } = this.state;

    const {
      name,
      image,
      level,
    } = data;

    return (
      <li className="item">
        <div className="icon icon--inventory icon--flch-dummied">
          <img
            alt={name}
            onMouseLeave={this.handleMouseLeave}
            onMouseMove={this.handleMouseMove}
            ref={(element) => { this.element = element; }}
            src={`${IMAGE_ROOT}/${image}small.png`}
          />
          <span className="js-item-value icon__value">{level}</span>
        </div>
        {showToolTip && (<ToolTip data={data} parent={this.element} active={showToolTip} />)}
      </li>
    );
  }
}

DummiedItem.propTypes = {
  data: PropTypes.object.isRequired, // eslint-disable-line
};
