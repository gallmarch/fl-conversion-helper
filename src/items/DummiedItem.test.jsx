import React from 'react';
import renderer from 'react-test-renderer';

import DummiedItem from './DummiedItem';

describe('DummiedItem', () => {
  it('matches its snapshot', () => {
    const inventoryMatch = createMockInventoryMatch();
    const id = 'expected-id';
    const quantity = 5;
    const props = { id, inventoryMatch, quantity };

    const component = renderer.create(<DummiedItem {...props} />);
    expect(component).toMatchSnapshot();
  });
});

function createMockInventoryMatch() {
  const container = document.createElement('div');
  const img = document.createElement('img');
  img.setAttribute('src', '//example.com/image.png');
  img.setAttribute('alt', 'some alt text');
  container.appendChild(img);
  return container;
}
