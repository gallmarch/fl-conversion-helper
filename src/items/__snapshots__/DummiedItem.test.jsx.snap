// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DummiedItem matches its snapshot 1`] = `
<li>
  <a
    className="tooltip flch-unlock"
  >
    <div
      className="qq"
    >
      5
    </div>
    <div>
      <img
        alt="some alt text"
        height={null}
        src="//example.com/image.png"
        width={null}
      />
    </div>
    <span
      className="tt"
    />
  </a>
</li>
`;
